### ISSUE TITLE
Label feature doesn't support Chinese as key?

### ISSUE THREAD

**Issue by juzhiyuan** [apache/apisix-dashboard#1300]
# Bug report

## Describe the bug

When I create a route with Label, an error occurred when using Chinese as key.

## Screenshots

![image](https://user-images.githubusercontent.com/2106987/104607114-706d0c80-56bb-11eb-8c6b-6d0ac3181eb0.png)

![image](https://user-images.githubusercontent.com/2106987/104606805-1c622800-56bb-11eb-9536-be01b06b92de.png)

## System information

- Version: master branch

## Additional context

```sh
$ curl 'http://139.217.190.60/apisix/admin/routes/337358106080052037' \
  -X 'PUT' \
  -H 'Accept: application/json' \
  --data-binary '{"name":"route_with_servicenoname","status":1,"methods":["GET","HEAD","POST","PUT","DELETE","OPTIONS","PATCH"],"priority":0,"uris":["/*"],"labels":{"分组1":"标签2","API_VERSION":"v1"},"vars":[],"upstream_id":"337065625484526491"}' 
```

**Comment by juzhiyuan** [comment]
Chinese as the key is a normal requirement IMHO, what's your opinion?

cc @liuxiran @nic-chen @membphis

**Comment by imjoey** [comment]
@juzhiyuan

Chinese characters as the name or value of Label is not such common. Until Kubernetes 1.17+, the name and value of Label are strictly limited to match `'([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]`, not accepting Chinese. IMHO,  we could keep the same as Kubernetes. 

Looking forward to others insights. Thanks.

**Comment by juzhiyuan** [comment]
but this is used as a label's key 🤔

**Comment by imjoey** [comment]
> @juzhiyuan
> 
> Chinese characters as the name or value of Label is not such common. Until Kubernetes 1.17+, the name and value of Label are strictly limited to match `'([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]`, not accepting Chinese. IMHO, we could keep the same as Kubernetes.
> 
> Looking forward to others insights. Thanks.

> but this is used as a label's key 🤔

@juzhiyuan sorry for not clear enough. The `name` mentioned above is the `key` of Label.

**Comment by liuxiran** [comment]
Technical aspects aside, I would vote for supporting Chinese labels 🤦‍♀️, both the key and value, that would be convenient for Chinese users

**Comment by juzhiyuan** [comment]
let's wait for others' opinion 💯

**Comment by starsz** [comment]
I think we should support Chinese labels. This will be more convenient for some companies.
In another way, we only need to update the validate schema, then we can support the Chinese labels., why not?

**Comment by imjoey** [comment]
Yep, I slightly changed my mind and we should confirm the following two things fine, then we can add support.

- If Chinese is supported, it should be better to also support all other non English languages
- As APISIX can be deployed in kubernetes as ingress controller at present, maybe as sidecar in future, we shall ensure that this label will not introduce the incompatibility with kubernetes.

Thanks.

**Comment by juzhiyuan** [comment]
thanks for your point! @gxthrj please take a look at this issue, too.

**Comment by membphis** [comment]
> Chinese as the key is a normal requirement IMHO, what's your opinion?
> * If Chinese is supported, it should be better to also support all other non English languages

agree to support Chinese and other non-English languages.

**Comment by juzhiyuan** [comment]
failed with English.

```sh
curl 'http://139.217.190.60/apisix/admin/routes' \
  -H 'Content-Type: application/json;charset=UTF-8' \
  --data-binary '{"name":"test_route_with_label","desc":"","status":1,"methods":["GET","HEAD","POST","PUT","DELETE","OPTIONS","PATCH"],"priority":0,"uris":["/*"],"labels":{"Group1":"测试 API","Group2":"示例 API"},"vars":[],"upstream_id":"337065625484526491"}' 
```

```json
{"code":10000,"message":"schema validate failed: labels.Group1: Does not match pattern '^[a-zA-Z0-9-_.]+$'\nlabels.Group2: Does not match pattern '^[a-zA-Z0-9-_.]+$'"}
```

**Comment by gxthrj** [comment]
Agree, I did not find a suitable reason to refuse to use Chinese as the key 👍

**Comment by tokers** [comment]
Agree + 1, as long as we have a strong mechanism to detect malformed label, it's no problem to accept any language.

**Comment by juzhiyuan** [comment]
yep, then who is glad to take this issue?

**Comment by nic-chen** [comment]
need support in APISIX first if we want to support it

**Comment by membphis** [comment]
do we need to transfer this issue to APISIX?

**Comment by nic-chen** [comment]
cc @spacewander
